dev-tools/check_duplicate_test_data.php

```diff
@@ -0,0 +1,201 @@
+use PhpCsFixer\Tests\Fixer\ConstantNotation\NativeConstantInvocationFixerTest;
+use PhpCsFixer\Tests\Fixer\ControlStructure\NoBreakCommentFixerTest;
+use PhpCsFixer\Tests\Fixer\ControlStructure\NoUnneededControlParenthesesFixerTest;
+use PhpCsFixer\Tests\Fixer\ControlStructure\NoUselessElseFixerTest;
+use PhpCsFixer\Tests\Fixer\ControlStructure\YodaStyleFixerTest;
+use PhpCsFixer\Tests\Fixer\DoctrineAnnotation\DoctrineAnnotationArrayAssignmentFixerTest;
+use PhpCsFixer\Tests\Fixer\DoctrineAnnotation\DoctrineAnnotationSpacesFixerTest;
+use PhpCsFixer\Tests\Fixer\FunctionNotation\PhpdocToParamTypeFixerTest;
+use PhpCsFixer\Tests\Fixer\Import\OrderedImportsFixerTest;
+use PhpCsFixer\Tests\Fixer\LanguageConstruct\SingleSpaceAfterConstructFixerTest;
+use PhpCsFixer\Tests\Fixer\LanguageConstruct\SingleSpaceAroundConstructFixerTest;
+use PhpCsFixer\Tests\Fixer\Operator\StandardizeIncrementFixerTest;
+use PhpCsFixer\Tests\Fixer\Phpdoc\NoSuperfluousPhpdocTagsFixerTest;
+use PhpCsFixer\Tests\Fixer\Phpdoc\PhpdocInlineTagNormalizerFixerTest;
+use PhpCsFixer\Tests\Fixer\Phpdoc\PhpdocTypesOrderFixerTest;
+use PhpCsFixer\Tests\Fixer\PhpUnit\PhpUnitStrictFixerTest;
+use PhpCsFixer\Tests\Fixer\PhpUnit\PhpUnitTargetVersionTest;
+use PhpCsFixer\Tests\Fixer\StringNotation\SingleQuoteFixerTest;
+use PhpCsFixer\Tests\Fixer\Whitespace\MethodChainingIndentationFixerTest;
+use PhpCsFixer\Tests\Fixer\Whitespace\SpacesInsideParenthesesFixerTest;
+use PhpCsFixer\Tests\FixerNameValidatorTest;
+use PhpCsFixer\Tests\Tokenizer\Analyzer\AlternativeSyntaxAnalyzerTest;
+use PhpCsFixer\Tests\Tokenizer\Analyzer\AttributeAnalyzerTest;
+use PhpCsFixer\Tests\Tokenizer\Analyzer\ClassyAnalyzerTest;
+use PhpCsFixer\Tests\Tokenizer\Analyzer\FunctionsAnalyzerTest;
+use PhpCsFixer\Tests\Tokenizer\TokensAnalyzerTest;
+use PhpCsFixer\Tests\Tokenizer\TokenTest;
+use PhpCsFixer\Tests\Tokenizer\Transformer\ReturnRefTransformerTest;
+use PhpCsFixer\Tests\UtilsTest;
+
+require_once __DIR__.'/../vendor/autoload.php';
+
+$testClassNames = array_filter(
+    array_keys(require __DIR__.'/../vendor/composer/autoload_classmap.php'),
+    static fn (string $className): bool => str_starts_with($className, 'PhpCsFixer\Tests\\')
+);
+
+if ([] === $testClassNames) {
+    echo 'Run: composer dump-autoload --optimize --working-dir=', realpath(__DIR__.'/..'), PHP_EOL;
+
+    exit(1);
+}
+
+$duplicatesFound = false;
+
+foreach ($testClassNames as $testClassName) {
+    if (!str_ends_with($testClassName, 'Test')) {
+        continue;
+    }
+
+    foreach ((new ReflectionClass($testClassName))->getMethods() as $method) {
+        if (!$method->isPublic()) {
+            continue;
+        }
+        if ($method->getDeclaringClass()->getName() !== $testClassName) {
+            continue;
+        }
+        if (!str_starts_with($method->getName(), 'provide')) {
+            continue;
+        }
+
+        $exceptions = array_map(
+            static function (Closure $dataProvider): string {
```

Comment:
@mvorisek that's the best I've come up with, do you have something better/shorter?

Reply:
Nevermind,

> First-class callable syntax is only allowed since PHP 8.1 

We need to wait a little while to use it.